include/seqan3/argument_parser/argument_parser.hpp

```diff
@@ -0,0 +1,508 @@
+     *
+     *~~~~~~~~~~~~~~~{.cpp}
+     *     int main(int argc, char ** argv)
+     *     {
+     *          seqan3::argument_parser myparser(argc, argv); // initialize
+     * 
+     *          myparser.info.app_name = "Penguin_Parade";
+     *          myparser.info.version = "2.0.0";
+     *          myparser.info.short_description = "Organize your penguin parade";
+     *          myparser.info.description.push_back("First Paragraph.");
+     *          myparser.info.description.push_back("Second Paragraph.");
+     *          myparser.info.example.push_back("./penguin_parade -d 10 -m 02 -y 2017 --number_of_penguins 100");
+     * 
+     *          return 0;
+     *     }
+     *~~~~~~~~~~~~~~~
+     *
+     * \cond DEV
+     * The meta information are needed in all the format classes (parse_format,
+     * help_format, ...etc) and the argument_parser, so they can't be part of the 
+     * argument_parser class itself but need to be independent.
+     * \endcond
+     *
+     */
+    meta_information info;
+
+// #ifdef SEQAN_VERSION_CHECK_OPT_IN
+//     bool  enabled_version_check{false};
+// #else  // Make version update opt out.
+//     bool  enabled_version_check{true};
+// #endif  // SEQAN_VERSION_CHECK_OPT_IN
+    // std::future<bool> appVersionCheckFuture;
+
+private:
+    //!\privatesection
+
+    /*!\brief Initilizes the argument_parser class on construction
+     *
+     * \throws seqan3::parser_invalid_argument
+     *
+     * \details
+     * This function adds all command line parameters to a std::vector<std::string>
+     * to take advantage of the vector functionality later on. Additionally,
+     * the format member variable is set, depending on which parameters are given
+     * by the user:
+     *
+     * - **-h/--help** sets the format to seqan3::detail::help_format
+     * - **-hh/--advanced-help** sets the format to seqan3::detail::help_format
+     *                           and show_advanced_options to `true`.
+     * - **-export-help html** sets the format to seqan3::detail::html_format
+     * - **-export-help man** sets the format to seqan3::detail::man_format
+     * - **-export-help ctd** sets the format to seqan3::detail::ctd_format
+     * - else the format is that to seqan3::detail::parse_format
+     *
+     * If `-export-help` is specified with a value other than html/man or ctd
+     * a parser_invalid_argument is thrown.
+     */
+    void init(int argc, char ** argv)
+    {
+        for(int i = 1; i < argc; ++i) // start at 1 to skip binary name
+        {
+            std::string arg{argv[i]};
+            argv_.push_back(arg);
+
+            if (arg == "-h" || arg == "--help")
+            {
+                format = detail::help_format(false);
+                return;
+            }
+            else if (arg == "-hh" || arg == "--advanced-help")
+            {
+                format = detail::help_format(true);
+                return;
+            }
+            else if (arg == "--export-help")
```

Comment:
Hm, we could let them inherit from help_format, but overwrite the `parse()` function to ignore all  add_*() calls and instead just print the version info, etc.
I actually like it but don't you think it's a little over-engineered for just printing out `"version 3.2.2"`?

Side question: When a class inherits a public function, which uses a private (parent) class member, this function will not work for the child-class right? I would need to make the corresponding members protected?